### PR TITLE
Provides SSL/TLS support within the API

### LINKED ISSUE #45: SSL/TLS Support within the API itself
Support for SSL/TLS within the API itself is available. 
Make it configurable so it's available for use.
This makes setup less complicated: No need for an additional proxy and SSL termination.

### PR DESCRIPTION
**Pull request by pj-spoelders**
Closes #45 
Provides SSL/TLS support within the API

### CODE REVIEW

File: doc/user_guide/user-guide.md [hunk 1]
```diff
@@ -100,8 +103,17 @@ The service allows 30 requests per minute for all API endpoints. The limitation
 
 ### Using Secured Connection (HTTPS)
 
-We strongly recommend setting up an HTTPS when deploying the API service.
-A recommended way is setting up a publicly accessible proxy server that [terminates SSL](https://en.wikipedia.org/wiki/TLS_termination_proxy) and forwards the request to the API service.
+We strongly recommend using HTTPS when deploying the API service.
+
+#### Enable TLS within the API itself
+
+You can enable HTTPS within the service itself using the API_TLS (set it to true to enable TLS), API_TLS_PKPATH (path to private key.perm file) and API_TLS_CERTPATH (path to certificate.pem file) configuration properties.
```
Comment (kaklakariada) [inline doc/user_guide/user-guide.md:110]:
```suggestion
You can enable HTTPS within the service itself using the `API_TLS` (set it to true to enable TLS), API_TLS_PKPATH (path to private `key.pem` file) and `API_TLS_CERTPATH` (path to `certificate.pem` file) configuration properties.
```

File: doc/user_guide/user-guide.md [hunk 2]
```diff
@@ -100,8 +103,17 @@ The service allows 30 requests per minute for all API endpoints. The limitation
...
+
+![image-20211201165901918](user-guide.assets/image-20211201165901918.png)
```
Comment (kaklakariada) [inline doc/user_guide/user-guide.md:112]:
```suggestion

![image-20211201165901918](user-guide.assets/image-20211201165901918.png)
```
There is no picture in Git

File: pkg/exasol-rest-api/application_runner.go [hunk 3]
```diff
@@ -26,7 +27,12 @@ func Run() {
 	swaggerURL := ginSwagger.URL("/swagger/doc.json")
 	AddEndpoints(router, application)
 	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler, swaggerURL))
-	err := router.Run(applicationProperties.ApplicationServer)
+	var err error
+	if applicationProperties.APIUseTLS {
+		err = router.RunTLS(applicationProperties.ApplicationServer, applicationProperties.APITLSCertificatePath, applicationProperties.APITLSPrivateKeyPath)
```
Comment (kaklakariada) [inline pkg/exasol-rest-api/application_runner.go:32]: Please add an integration test for HTTPS or create a ticket to do this later.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: doc/user_guide/user-guide.md (hunk 2 above)
Comment (pj-spoelders) [reply doc/user_guide/user-guide.md:112]: Included the picture

File: pkg/exasol-rest-api/application_runner.go (hunk 3 above)
Comment (pj-spoelders) [reply pkg/exasol-rest-api/application_runner.go:32]: See https://github.com/exasol/exasol-rest-api/issues/47

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**